extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -177,27 +184,160 @@ export class MigrationStateModel implements Model, vscode.Disposable {
+					'platform': this._assessmentApiResponse.assessmentResult.serverHostPlatform,
+					'engineEdition': this._assessmentApiResponse.assessmentResult.serverEngineEdition
+				},
+				{
+					'issuesCount': this._assessmentResults.issues.length,
+					'warningsCount': this._assessmentResults.databaseAssessments.reduce((count, d) => count + d.issues.length, 0),
+					'duration': endTime - startTime,
+					'databaseCount': this._assessmentResults.databaseAssessments.length,
+					'serverHostCpuCount': this._assessmentApiResponse.assessmentResult.cpuCoreCount,
+					'serverHostPhysicalMemory': this._assessmentApiResponse.assessmentResult.physicalServerMemory,
+					'serverDatabases': this._assessmentApiResponse.assessmentResult.numberOfUserDatabases,
+					'serverDatabasesReadyForMigration': this._assessmentApiResponse.assessmentResult.sqlManagedInstanceTargetReadiness.numberOfDatabasesReadyForMigration,
+					'offlineDatabases': this._assessmentApiResponse.assessmentResult.sqlManagedInstanceTargetReadiness.numberOfNonOnlineDatabases
+				}
+			);
+
+			const databaseWarningsMap: Map<string, number> = new Map();
+			const databaseErrorsMap: Map<number, number> = new Map();
+
+			this._assessmentApiResponse.assessmentResult.databases.forEach(d => {
+
+				sendSqlMigrationActionEvent(
+					TelemetryViews.MigrationWizardTargetSelectionPage,
+					TelemetryAction.DatabaseAssessment,
+					{
+						'sessionId': this._sessionId,
+						'hashedDatabaseName': hashString(d.name),
+						'compatibilityLevel': d.compatibilityLevel
+					},
+					{
+						'warningsCount': d.items.length,
+						'errorsCount': d.errors.length,
+						'assessmentTime': d.assessmentTimeInMilliseconds,
+						'numberOfBlockerIssues': d.sqlManagedInstanceTargetReadiness.numOfBlockerIssues,
+						'databaseSize': d.databaseSize
+					}
+				);
+
+				d.items.forEach(i => {
+					if (databaseWarningsMap.has(i.ruleId)) {
+						databaseWarningsMap.set(i.ruleId, databaseWarningsMap.get(i.ruleId)! + i.impactedObjects.length);
+					} else {
+						databaseWarningsMap.set(i.ruleId, i.impactedObjects.length);
+					}
+				});
+
+				d.errors.forEach(e => {
+					if (databaseErrorsMap.has(e.errorId)) {
+						databaseErrorsMap.set(e.errorId, databaseErrorsMap.get(e.errorId)! + 1);
+					} else {
+						databaseErrorsMap.set(e.errorId, 1);
+					}
+				});
+
+			});
+
+			databaseWarningsMap.forEach((v, k) => {
+				sendSqlMigrationActionEvent(
+					TelemetryViews.MigrationWizardTargetSelectionPage,
+					TelemetryAction.DatabaseAssessmentWarning,
+					{
+						'sessionId': this._sessionId,
+						'ruleId': k
+					},
+					{
+						'count': v
+					}
+				);
+			});
+
+			databaseErrorsMap.forEach((v, k) => {
+				sendSqlMigrationActionEvent(
+					TelemetryViews.MigrationWizardTargetSelectionPage,
+					TelemetryAction.DatabaseAssessmentError,
+					{
+						'sessionId': this._sessionId
+					},
+					{
+						'errorId': k,
```

Comment:
The errorId isn't really a measurement, this should probably be in the properties bucket